src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,128 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int GROWTH_AFFIX = 1;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (isNotValidToAdd(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Non existent "
+                    + "position: " + index);
+        }
+        grow();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (isNotValidToFind(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Not found "
+                    + "element at position: ");
+        } else {
+            return array[index];
+        }
```

Comment:
- you don't need else after returning from function (this means throwing too)
- put throwing inside of private method, since it's a repeated part too